hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -274,17 +278,16 @@ data PluginDescriptor =
                    } deriving (Generic)
 
 instance Show PluginCommand where
-  show (PluginCommand name _ _) = "PluginCommand { name = " ++ T.unpack name ++ " }"
-
-type PluginId = T.Text
-type CommandName = T.Text
+  show (PluginCommand i _) = "PluginCommand { name = " ++ show i ++ " }"
 
-newtype CommandFunc a b = CmdSync (a -> IdeGhcM (IdeResult b))
+newtype CommandId = CommandId T.Text
+  deriving (Show, Read, Eq, Ord)
+instance IsString CommandId where
+  fromString = CommandId . T.pack
 
 data PluginCommand = forall a b. (FromJSON a, ToJSON b, Typeable b) =>
-  PluginCommand { commandName :: CommandName
-                , commandDesc :: T.Text
-                , commandFunc :: CommandFunc a b
```

Comment:
Why is the description removed and the `CommandFunc`? Are these remnants from JsonStdio? Is the description displayed somewhere for lsp-clients?

Reply:
Descriptions and name were only used for the plugin querying commands in base, which aren't accessible through LSP anymore. `PluginCommand` should now be isomorphic to the LSP commands

Reply:
I removed `CommandFunc` just to simplify things. I think it's more illustrative to have the function type directly visible inside `PluginCommand` and it removes the redundant

```haskell
xCmd :: CmdFunc a b
```
```haskell
xCmd' :: a -> IdeGhcM (IdeResult b)
```

pattern